test/unit/prompts/ChannelCaptureContainer/ChannelCaptureContainer.ts

```diff
@@ -0,0 +1,117 @@
+  await TestEnvironment.initialize(options);
+});
+
+test.afterEach(function () {
+  sandbox.restore();
+});
+
+const testHelper = new SlidedownPromptingTestHelper(sandbox);
+const minimalSmsAndEmailOptions = testHelper.getMinimalSmsAndEmailOptions();
+
+test("check that calling mount() adds the capture container to DOM", async t => {
+  ChannelCaptureContainerHelper.setupStubs(sandbox);
+  const captureContainer = new ChannelCaptureContainer(minimalSmsAndEmailOptions);
+  let containerFromDom = getDomElementOrStub(`#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.channelCaptureContainer}`);
+  t.false(containerFromDom.id === CHANNEL_CAPTURE_CONTAINER_CSS_IDS.channelCaptureContainer);
+
+  await captureContainer.mount();
+
+  containerFromDom = getDomElementOrStub(`#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.channelCaptureContainer}`);
+  t.true(containerFromDom.id === CHANNEL_CAPTURE_CONTAINER_CSS_IDS.channelCaptureContainer);
+});
+
+test("sms & email validation is hidden by default", async t => {
+  ChannelCaptureContainerHelper.setupStubs(sandbox);
+
+  const captureContainer = new ChannelCaptureContainer(minimalSmsAndEmailOptions);
+  await captureContainer.mount();
+
+  const smsValidationElement = document.querySelector(
+    `#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalSmsValidationElement}`
+  );
+
+  const smsValidationHidden = smsValidationElement?.classList.contains(
+    CHANNEL_CAPTURE_CONTAINER_CSS_CLASSES.onesignalValidationElementHidden
+    );
+
+  const emailValidationElement = document.querySelector(
+    `#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalEmailValidationElement}`
+  );
+
+  const emailValidationHidden = emailValidationElement?.classList.contains(
+    CHANNEL_CAPTURE_CONTAINER_CSS_CLASSES.onesignalValidationElementHidden
+    );
+
+  t.true(smsValidationHidden);
+  t.true(emailValidationHidden);
+});
+
+test("resetting validation styles works correctly", async t => {
+  ChannelCaptureContainerHelper.setupStubs(sandbox);
+
+  const captureContainer = new ChannelCaptureContainer(minimalSmsAndEmailOptions);
+  await captureContainer.mount();
+
+  ChannelCaptureContainer.showSmsInputError(true);
+  ChannelCaptureContainer.showEmailInputError(true);
+
+  ChannelCaptureContainer.resetInputErrorStates(DelayedPromptType.SmsAndEmail);
+
+  const smsValidationElement = document.querySelector(
+    `#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalSmsValidationElement}`
+  );
+
+  const smsValidationHidden = smsValidationElement?.classList.contains(
+    CHANNEL_CAPTURE_CONTAINER_CSS_CLASSES.onesignalValidationElementHidden
+    );
+
+  const emailValidationElement = document.querySelector(
+    `#${CHANNEL_CAPTURE_CONTAINER_CSS_IDS.onesignalEmailValidationElement}`
+  );
+
+  const emailValidationHidden = emailValidationElement?.classList.contains(
+    CHANNEL_CAPTURE_CONTAINER_CSS_CLASSES.onesignalValidationElementHidden
+    );
+
+  t.true(smsValidationHidden);
+  t.true(emailValidationHidden);
+});
```

Comment:
check the error state / validation working correctly

Reply:
was only able to check for email since the validation for sms is outsourced to the 3rd party library loaded via script